File: interfaces/java/com/kx/Minute.java
package com.kx;

import static com.kx.Type.NullInteger;
import static com.kx.Type.toTwoDigitString;

/**
 * This class is a kdb+ minute.
 * 
 * @see <a href="https://code.kx.com/trac/wiki/Cookbook/InterfacingWithJava">Interfacing With Java at code.kx</a> 
 * @author tech@kx.com
 */
public class Minute {
	
	private int minute;

	/**
	 * Constructs the minute.
	 * 
	 * @param minute
	 */
	public Minute(int minute) {
		this.minute = minute;
	}

	/**
 	 * Converts to a string.
 	 *
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		if (minute == NullInteger)
			return "";
		else
			return toTwoDigitString(minute / 60) 
					+ ":" 
					+ toTwoDigitString(minute % 60);
	}

	/**
         * Checks for equality.
         *
	 * @see java.lang.Object#equals(Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (obj instanceof Minute) 
			return (minute == ((Minute)obj).minute);
		else 
			return false;
	}

	/**
	 * Gets the hashcode.
	 *
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		return minute;
	}

	/**
	 * Gets the minute.
	 * 
	 * @return the minute
	 */
	public int getMinute() {
		return minute;
	}

	/**
	 * Sets the minute.
	 * 
	 * @param minute the new value
	 */
	public void setMinute(int minute) {
		this.minute = minute;
	}
}

File: interfaces/csharp/Kdbplus/Minute.cs
namespace Kdbplus
{
    /// <summary>
    /// This is a kdb+ minute value.
    /// </summary>
    public class Minute
    {
        private int theMinute;

        /// <summary>
        /// Constructs based on the given int value.
        /// </summary>
        /// <param name="m">The months.</param>
        public Minute(int m)
        {
            theMinute = m;
        }

        /// <summary>
        /// Gets and sets the minute value.
        /// </summary>
        public int TheMinute
        {
            get { return theMinute; }
            set { theMinute = value; }
        }

        /// <summary>
        /// Converts to a string.
        /// </summary>
        /// <returns>The string representation of this value.</returns>
        public override string ToString()
        {
            return theMinute == Type.NullInteger ? "" : Type.ToTwoDigitString(theMinute/60) + ":" + Type.ToTwoDigitString(theMinute%60);
        }
    }
}